src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -300,6 +300,21 @@ end
     return T
 end
 
+@inline function enstrophy(u, gradients, equations::CompressibleNavierStokesDiffusion2D)
+    # Enstrophy is 0.5 rho ω⋅ω where ω = ∇ × v
+
+    omega = vorticity(u, gradients, equations)
+    return 0.5 * u[1] * omega^2
+end
+
+@inline function vorticity(u, gradients, equations::CompressibleNavierStokesDiffusion2D)
+    # Ensure that we have velocity `gradients` by way of the `convert_gradient_variables` function.
+    _, dv1dx, dv2dx, _ = convert_derivative_to_primitive(u, gradients[1], equations)
+    _, dv1dy, dv2dy, _ = convert_derivative_to_primitive(u, gradients[2], equations)
```

Comment:
Does this assume that the gradients are computed with respect to the conservative variables? If so, we should specialize the dispatch on `equations` to the appropriate type parameter.

Reply:
As far as I understand this, the type (conserved/primitive) of the gradients is determined by the type in which they are stored in `cache_parabolic`, see the existing version for 3D:

https://github.com/trixi-framework/Trixi.jl/blob/fe0e78c658283db21e581bbbc6d48d0adcf39510/src/callbacks_step/analysis_dg3d.jl#L230-L248

Reply:
You're right, I missed that. Thanks!